### PR TITLE
SDIT-1933 Raise "receive" event for move booking

### PR DESCRIPTION
**Pull request by andymarke**
When a NOMIS move booking is used for an active booking that is moved to a newly created prisoner, last movement could be any movement but previous booking will be null

Ktlint converting to expression body

### CODE REVIEW

File: hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/events/PrisonerMovementsEventService.kt [hunk 1]
```diff
@@ -62,49 +62,49 @@ class PrisonerMovementsEventService(
...
+      TransferOut(prisonerNumber, previousPrisonerSnapshot?.prisonId!!)
+    } else if (prisoner.isCourtOutMovement(previousPrisonerSnapshot)) {
+      SentToCourt(prisonerNumber, previousPrisonerSnapshot?.prisonId!!)
+    } else if (prisoner.isTAPOutMovement(previousPrisonerSnapshot)) {
+      TAPRelease(prisonerNumber, previousPrisonerSnapshot?.prisonId!!)
+    } else if (prisoner.isReleaseToHospital(previousPrisonerSnapshot)) {
+      ReleasedToHospital(prisonerNumber, previousPrisonerSnapshot?.prisonId!!)
+    } else if (prisoner.isRelease(previousPrisonerSnapshot)) {
+      Released(prisonerNumber, previousPrisonerSnapshot?.prisonId!!)
+    } else if (prisoner.isNewAdmissionDueToMoveBooking(previousPrisonerSnapshot)) {
```
Comment (andymarke) [inline hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/events/PrisonerMovementsEventService.kt:95]: The actual change - prisoner.isNewAdmissionDueToMoveBooking

File: hmpps-prisoner-search-indexer/src/test/resources/receive-state-changes/first-moved-booking.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  "prisonerNumber": "A9460DY",
+  "bookingId": "1203208",
+  "bookNumber": "40210A",
+  "firstName": "AKUSEA",
+  "lastName": "BOATENG",
+  "dateOfBirth": "1970-01-01",
+  "gender": "Female",
+  "youthOffender": false,
+  "status": "ACTIVE IN",
+  "lastMovementTypeCode": "CRT",
```
Comment (andymarke) [inline hmpps-prisoner-search-indexer/src/test/resources/receive-state-changes/first-moved-booking.json:11]: last movement for a move booking could be anything not just an ADM

File: hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/events/PrisonerMovementsEventService.kt [hunk 3]
```diff
@@ -62,49 +62,49 @@ class PrisonerMovementsEventService(
     previousPrisonerSnapshot: Prisoner?,
     prisoner: Prisoner,
     offenderBooking: OffenderBooking,
-  ): PossibleMovementChange {
```
Comment (andymarke) [inline hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/events/PrisonerMovementsEventService.kt:65]: ktlint is converted to expression body - see actual change below

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**